src/Microsoft.OData.Core/ODataWriterCore.cs

```diff
@@ -2561,10 +2561,10 @@ private void EnterScope(WriterState newState, ODataItem item)
             IEdmNavigationSource navigationSource = null;
             IEdmType itemType = null;
             SelectedPropertiesNode selectedProperties = currentScope.SelectedProperties;
-            ODataUri odataUri = currentScope.ODataUri.Clone();
-            if (odataUri.Path == null)
+            ODataUriSlim odataUriSlim = new ODataUriSlim(currentScope.ODataUri);

```

Comment:
```suggestion
            ODataUriSlim odataUri = new ODataUriSlim(currentScope.ODataUri);
```
I wish you could retain the variable name as `odataUri`. It'd even makes your diff smaller

Reply:
I thought using `odataUriSlim` would make the changes and intentions more clear, and reduce confusion if we have both `ODataUri` and `ODataUriSlim` instances in the same scope. But I'm open to changing if you feel like it's better.